include/hacks/__availability

```diff
@@ -0,0 +1,27 @@
+/**
+ * This is a hack to make RGBDS build on macOS 10.14 and below. Without it, we get an error:
+ *     error: 'path' is unavailable: introduced in macOS 10.15
+ * The hack weakens the `std::filesystem::path` availability error into a warning, by removing
+ * `strict` from the availability macros (for information on the `availability` attribute, see
+ * https://releases.llvm.org/17.0.1/tools/clang/docs/AttributeReference.html#availability). This is
+ * acceptable because what we need of `std::filesystem::path` was already supported in macOS 10.14
+ * (the last version to support 32-bit, hence in demand despite being superseded by 10.15 in 2019).
+ */
+
+#include_next <__availability>
```

Comment:
Should we do a typical `#ifndef RGBDS_AVAILABILITY_H` anyway? Also `/* SPDX-License-Identifier: MIT */`.

Reply:
No, only our CI macOS build sees this header.

`__availability`, as the two leading underscores hint at, is a header internal to macOS. Its `#include` follows the usual header resolution process, but since [even `-I` directories are searched for system headers](https://gcc.gnu.org/onlinedocs/gcc/Directory-Options.html), `include/hacks/__availability` is found first.

It [continues the search](https://gcc.gnu.org/onlinedocs/cpp/Wrapper-Headers.html), thus processing the actual system header; but then we tweak some of the macros it defines to suit our needs.

Think of it as interposition between two parts of the macOS libc. Yes, it's a hack (I don't know of a single use of `#include_next` that isn't), but we're *aiming* to play outside of Apple's rules, so there won't be a clean way to do what we're looking for.

As to your question of whether we should add an include guard, sure.

Reply:
My concern is, shouldn't a user building rgbds themselves on macOS still get this `include`? Which afaik means putting `-I include/hacks` in the Makefile itself, not just our own CI builds.

Reply:
No, this is only necessary when building static macOS binaries, which shouldn't be what the average macOS user will want...

That said, the way you put this, I realise that we should actually test **both** configurations: "standard" macOS build (with `develop` checks), *and* static binaries.

This would mean keeping macOS in the `unix-testing` path, and also not using `develop` in the latter (thereby removing the roadblock to merging this, and also harmonising the "test" and "release" configs).

How does that sound?

Reply:
So we'd have `unix-testing` for Linux and non-`develop` static macOS (to ensure its compatibility); and `macos-testing` for `develop` macOS (to get all those extra `develop` checks). Correct? Sounds good.

Reply:
Actually, the two macOS targets would be the opposite: `unix-testing` has `develop` (like it currently does), and a new `static_macos_testing` flow for, well,